### ISSUE TITLE
[Breaking][Xcode 13.0 b3]: "error: 'shared' is unavailable in application extensions for iOS:"

### ISSUE THREAD

**Issue by Panajev** [firebase/firebase-ios-sdk#8401]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.0 b3
  * Firebase SDK version: 8.3.1
  * Installation method: Swift Package Manager
  * Firebase Component: App Distribution
  
### [REQUIRED] Step 2: Describe the problem
Hello,

When compiling our application, I see an error like:

```
error: 'shared' is unavailable in application extensions for iOS: Use view controller based solutions where appropriate instead.
            if let windowScene = UIApplication.shared.activeScene {
```
Note that this was not happening in our app with Xcode 13.0 beta 2, it started happening today when I updated to Xcode 13.0 beta 3). Apple requires the library itself to be annotated like so:
```
@available(iOSApplicationExtension, unavailable)
```
See: https://forums.swift.org/t/set-application-extension-api-only-on-a-spm-package/39333/11 

This is related to a change in Xcode made by Apple deliberately and for which they provided annotation library authors have to add when declaring methods that would trigger that compilation error.

Thank you in advance for your help :)!

Kind Regards,

Goffredo Marocchi

**Notes**
* I reported the bug on Apple's Feedback Reporter tool (it affects lots of libraries and I have no way to let SwiftPM know we do not have any App Extension targets...): FB9337463
* Reported a thread in the dev forums too: https://developer.apple.com/forums/thread/685103 

#### Steps to reproduce:

What happened? How can we make the problem occur?
Download Xcode 13.0 b3, compile a project that imports Firebase iOS SDK (including App Distribution too) using Swift PM (cocoapods works for this library for now).

#### Relevant Code:
The compilation issue occurs in Firebase's code as the SwiftPM package is imported.

**Comment by Panajev** [comment]
<img width="816" alt="Screenshot 2021-07-17 at 11 27 57" src="https://user-images.githubusercontent.com/562991/126034079-4964c75a-1a2e-46e0-9e05-d5835b56f6ac.png">

**Comment by paulb777** [comment]
@Panajev Thanks for the report. The issue also shows up in Dynamic Links and In App Messaging.

It seems like the right solution for Swift libraries, is to add the `@available(iOSApplicationExtension, unavailable)` annotation, but it's not immediately clear what to do for Objective C libraries.

**Comment by Panajev** [comment]
Short short term one would get back to Cocoapods or Carthage for a while, but it is a royal pain. I did file a bug and I hope I can get some more dupes, they really help!

Thank you for being so on point and quick to answer on a weekend :)!

**Comment by paulb777** [comment]
Suggested approach at https://forums.swift.org/t/set-application-extension-api-only-on-a-spm-package/39333/21. I'll give it a try in the next few days if no one else beats me to it.

**Comment by paulb777** [comment]
Hi @Panajev We cherry-picked the fix to 8.4.0 so it should be out later this week.  In the meantime you could point to the release-8.4 branch.

**Comment by Panajev** [comment]
Thanks!!!!
